src/javascript/request-composer/request.js

```diff
@@ -0,0 +1,318 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import moment from 'moment';
+
+function buildReportRequest(params) {
+  let reportRequest = {
+      viewId: params.viewId,
+      samplingLevel: params.samplingLevel,
+  };
+  if (params.filters) {
+    reportRequest.filtersExpression = params.filters;
+  }
+  if (params.includeEmptyRows) {
+    if (params.includeEmptyRows.toLowerCase() == 'true') {
+      reportRequest.includeEmptyRows = 'true';
+    } else if (params.includeEmptyRows.toLowerCase() == 'false') {
+      reportRequest.includeEmptyRows = 'false';
+    }
+  }
+  if (params.pageSize &&
+      parseInt(params.pageSize)) {
+    reportRequest.pageSize = params.pageSize;
+  }
+  if (params.pageToken) {
+    reportRequest.pageToken = params.pageToken;
+  }
+  return reportRequest;
+}
+
+function applyDateRanges(request, params, settings) {
+  if (params.startDate &&
+      params.endDate &&
+      settings.requestType != 'COHORT') {
+    request.dateRanges = [{
+      'startDate': params.startDate,
```

Comment:
According the the style guide, object keys shouldn't use quotes unless they need to (e.g. they contain a space or hyphen). Can you fix this here and elsewhere?

I'll see if I can add this to the lint rules.